pkg/compute/publish.go

```diff
@@ -0,0 +1,98 @@
+package compute
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/text"
+)
+
+// PublishCommand produces and deploys an artifact from files on the local disk.
+type PublishCommand struct {
+	common.Base
+	manifest manifest.Data
+	build    *BuildCommand
+	deploy   *DeployCommand
+
+	// Deploy fields
+	path    common.OptionalString
+	version common.OptionalInt
+
+	// Build fields
+	name       common.OptionalString
+	lang       common.OptionalString
+	includeSrc common.OptionalBool
+	force      common.OptionalBool
+}
+
+// NewPublishCommand returns a usable command registered under the parent.
+func NewPublishCommand(parent common.Registerer, globals *config.Data, build *BuildCommand, deploy *DeployCommand) *PublishCommand {
+	var c PublishCommand
+	c.Globals = globals
+	c.manifest.File.SetOutput(c.Globals.Output)
+	c.manifest.File.Read(manifest.Filename)
+	c.build = build
+	c.deploy = deploy
+	c.CmdClause = parent.Command("publish", "Composite of \"build\", then \"deploy\"")
+
+	// Deploy flags
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("version", "Number of version to activate").Action(c.version.Set).IntVar(&c.version.Value)
+	c.CmdClause.Flag("path", "Path to package").Short('p').Action(c.path.Set).StringVar(&c.path.Value)
+
+	// Build flags
+	c.CmdClause.Flag("name", "Package name").Action(c.name.Set).StringVar(&c.name.Value)
+	c.CmdClause.Flag("language", "Language type").Action(c.lang.Set).StringVar(&c.lang.Value)
+	c.CmdClause.Flag("include-source", "Include source code in built package").Action(c.includeSrc.Set).BoolVar(&c.includeSrc.Value)
+	c.CmdClause.Flag("force", "Skip verification steps and force build").Action(c.force.Set).BoolVar(&c.force.Value)
+
+	return &c
+}
+
+// Exec implements the command interface.
+//
+// NOTE: unlike other non-aggregate commands that initialize a new
+// text.Progress type for displaying progress information to the user, we don't
+// use that in this command because the nested commands overlap the output in
+// non-deterministic ways. It's best to leave those nested commands to handle
+// the progress indicator.
+func (c *PublishCommand) Exec(in io.Reader, out io.Writer) (err error) {
+	// Reset the fields on the BuildCommand based on PublishCommand values.
+	if c.name.WasSet {
+		c.build.PackageName = c.name.Value
+	}
+	if c.lang.WasSet {
+		c.build.Lang = c.lang.Value
+	}
+	if c.includeSrc.WasSet {
+		c.build.IncludeSrc = c.includeSrc.Value
+	}
+	if c.force.WasSet {
+		c.build.Force = c.force.Value
+	}
+
+	err = c.build.Exec(in, out)
+	if err != nil {
+		return fmt.Errorf("error building package: %w", err)
```

Comment:
I don't think the message here is necessary and the same with the deploy one on line 94, as the sub-commands already have their own relevant messaging.